### DISCUSSION TITLE
academy/web-scraping-for-beginners/challenge/scraping-amazon

### DISCUSSION THREAD

**Discussion by giscus** [apify/apify-docs#752] [Comments]
# academy/web-scraping-for-beginners/challenge/scraping-amazon

Before you build your first web scraper with Crawlee, it is important to understand the concept of modularity in programming.

https://docs.apify.com/academy/web-scraping-for-beginners/challenge/scraping-amazon

**Comment by qshytpolite** [comment]
Nice tutorial, but this part fail with this error
WARN  CheerioCrawler: Reclaiming failed request back to the list or queue. 503 - Internal Server Error: <!--
        To discuss automated access to Amazon data please contact api-services-support@amazon.c
I am guessing we have to use playwright.

**Comment by Buyasser** [comment]
in the final code of routes.js it should read <import { BASE_URL, labels } from './constants.js';> i/o <import { BASE_URL, labels } from './constants';>
